src/Controller/Controller.php

```diff
@@ -497,16 +497,16 @@ public function shutdownProcess()
      *
      * @param string|array $url A string or array-based URL pointing to another location within the app,
      *     or an absolute URL
-     * @param int $status Optional HTTP status code (eg: 404)
+     * @param int $status Optional HTTP status code (eg: 301)
      * @return void|\Cake\Network\Response
      * @link http://book.cakephp.org/3.0/en/controllers.html#Controller::redirect
      */
-    public function redirect($url, $status = null)
+    public function redirect($url, $status = 302)
     {
         $this->autoRender = false;
 
         $response = $this->response;
-        if ($status && $response->statusCode() === 200) {
+        if ($status) {
```

Comment:
Maybe you can pass "null" manually to avoid this overwriting for some reason.
Or it is to protect from stuff like empty strings to avoid ending up calling this method?
I guess keeping that check on emptiness shouldn't hurt.